programs/stake/src/stake_state.rs

```diff
@@ -560,6 +560,14 @@ pub trait StakeAccount {
         stake_authorize: StakeAuthorize,
         signers: &HashSet<Pubkey>,
     ) -> Result<(), InstructionError>;
+    fn authorize_with_seed(
+        &self,
+        authority: &Pubkey,
+        stake_authorize: StakeAuthorize,
+        base: &KeyedAccount,
+        seed: &str,
+        address_owner: &Pubkey,
```

Comment:
It's not obvious to me what `address_owner` is supposed to be. How about naming it `old_authority` and renaming `authority` to `new_authority`? I also wouldn't mind if `base` and `seed` were named to match `old_authority` to be clear what they're for

Reply:
Also, what's the benefit of passing `address_owner`? Better error messaging? Seems like it only serves to ensure you passed the correct seed which I feel we already have enough information to verify.

Reply:
This is the `base, seed, owner` triplet passed to `Pubkey::new_with_seed`. Owner isn't an authority. It might be the system program id, for example. Sounds like naming those `authority_base, authority_seed, authority_owner` might help? Maybe also moving them to just after the `authority` argument too?

Reply:
Oh I see, didn't realize derived addresses were namespaced to a particular owner, yeah sounds good